### PR TITLE
CB-15873 CB-15874 CB-15875 CB-15879 Support Gov and Public AWS in the…

### PR DESCRIPTION
**Pull request by doktoric**
… same time for local dev. Modify mock idbroker mapping roles for gov cloud. Modify region mapping to include gov cloud regions. Add new crn segment to gov cloud deployment.

See detailed description in the commit message.

### CODE REVIEW

File: cloud-aws-gov/build.gradle
Comment (bergerdenes) [inline cloud-aws-gov/build.gradle:11]: duplicate

File: cloud-aws-gov/build.gradle
Comment (bergerdenes) [inline cloud-aws-gov/build.gradle:1]: missing jar name, version etc.

File: cloud-aws-gov/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsGovConnector.java
Comment (bergerdenes) [inline cloud-aws-gov/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsGovConnector.java:13]: is the AWS_GOV variant also NATIVE?
from `MultiAzValidator` class below I assume yes

File: core/build.gradle
Comment (bergerdenes) [inline core/build.gradle:60]: gov cloud is now here

File: core/src/main/java/com/sequenceiq/cloudbreak/service/identitymapping/AwsMockAccountMappingService.java
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/identitymapping/AwsMockAccountMappingService.java:51]: please refactor the multiple placeholder substitution by using some template handler, like https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringSubstitutor.html

File: environment/src/main/java/com/sequenceiq/environment/credential/service/CredentialService.java
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/credential/service/CredentialService.java:247]: please use value type `boolean`

File: environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/EnvironmentStorageConfigurationValidator.java
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/EnvironmentStorageConfigurationValidator.java:43]: `Amazon` or `AWS` (this is a user-facing message)

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java
Comment (doktoric) [reply cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java:73]: in case of aws I think it will be the same

File: cloud-aws-gov/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsGovConnector.java
Comment (doktoric) [reply cloud-aws-gov/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsGovConnector.java:13]: yes

**Review by bergerdenes — APPROVED** [review]

File: core/src/main/resources/application.yml
Comment (bergerdenes) [inline core/src/main/resources/application.yml:109]: just a side note: you will need a counterpart of this in many places like cbd, CSI etc.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/account/PreferencesService.java
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/account/PreferencesService.java:68]: Nit: Could be merged into a single `return` statement.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/account/PreferencesService.java
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/account/PreferencesService.java:65]: Perhaps replace with `StringUtils.isEmpty(enabledGovPlatforms)` to make it `null`-safe.

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java
Comment (lajosrodek) [inline cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java:72]: General note: There is a separate `AwsNativeCloudWatchService`. Would not it be better to reuse / modify that one instead?

**Review by lajosrodek — COMMENTED** [review]
There seem to be a few more `:aws:` occurrences, like `RoleBasedParameters.roleArn`, `AwsDiskEncryptionParameters` (`AWS_ENCRYPTION_KEY_INVALID_MESSAGE` and `REG_EXP_FOR_ENCRYPTION_KEY_ARN`), `EncryptionKeyArnValidator.validateEncryptionKeyArn()`.

**Comment by lacikaaa** [comment]
@lajosrodek it would be awesome if you could use change request instead of comments :)

File: core/src/main/resources/application.yml
Comment (doktoric) [reply core/src/main/resources/application.yml:112]: I know 👯

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java
Comment (lajosrodek) [reply cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java:72]: I fear `AwsNativeCloudWatchService.ALARM_ACTION_PREFIX` will be still wrong:
```
private static final String ALARM_ACTION_PREFIX = "arn:aws:automate:";
```

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/instance/alarm/AwsNativeCloudWatchService.java
Comment (lajosrodek) [inline cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/instance/alarm/AwsNativeCloudWatchService.java:211]: Nit: extra space after equals sign. :smile:

**Review by lajosrodek — APPROVED** [review]
Thanks for the fixes! LGTM

**Comment by doktoric** [comment]
please hold until csi changes are not merged

**Comment by lnardai** [comment]
Added do not merge until csi changes get resolved.

File: cloud-aws-cloudformation/src/main/resources/definitions/aws-zone-coordinates.json
Comment (biharitomi) [inline cloud-aws-cloudformation/src/main/resources/definitions/aws-zone-coordinates.json:117]: Couldn't this new gov cloud realted zones be stored in the gov cloud related module and variant?

Comment (doktoric) [reply cloud-aws-cloudformation/src/main/resources/definitions/aws-zone-coordinates.json:117]: I think this is also in common module so we should just remove the entirely file from here.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/imagecatalog/ImageCatalogV4Endpoint.java
Comment (biharitomi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/imagecatalog/ImageCatalogV4Endpoint.java:125]: Shouldn't it be more maintainable in the future if we introduce a variant field instead of the `govCloud` boolean?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/util/UtilV4Endpoint.java
Comment (biharitomi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/util/UtilV4Endpoint.java:55]: Shouldn't it be more maintainable in the future if we introduce a variant field instead of the govCloud boolean?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/imagecatalog/ImageCatalogV4Endpoint.java
Comment (doktoric) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/imagecatalog/ImageCatalogV4Endpoint.java:125]: we just discussed

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/util/UtilV4Endpoint.java
Comment (doktoric) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/util/UtilV4Endpoint.java:55]: we just discussed

**Review by biharitomi — APPROVED** [review]

**Comment by doktoric** [comment]
retest this please

**Pull request merged**
